File: packages/native/macos-native/macos/FRNPopUpButtonManager.m
#import <React/RCTComponent.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(FRNPopUpButtonManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(items, NSArray)
RCT_REMAP_VIEW_PROPERTY(selectedIndex, indexOfSelectedItem, NSInteger)
RCT_EXPORT_VIEW_PROPERTY(pullsDown, BOOL)
RCT_CUSTOM_VIEW_PROPERTY(disabled, BOOL, NSPopUpButton) {
  [view setEnabled:![RCTConvert BOOL:json]];
}
RCT_REMAP_VIEW_PROPERTY(tooltip, toolTip, NSString)
RCT_EXPORT_VIEW_PROPERTY(onChange, RCTBubblingEventBlock)

@end

File: packages/native/macos-native/macos/FRNPopUpButtonManager.swift
import AppKit
import Foundation
#if USE_REACT_AS_MODULE
import React
#endif // USE_REACT_AS_MODULE

@objc(FRNPopUpButtonManager)
class FRNPopUpButtonManager: RCTViewManager {

  override func view() -> NSView! {
    return FRNPopUpButton()
  }

  override class func requiresMainQueueSetup() -> Bool {
    return true
  }
}
